Automattic/wp-calypso
# Plans 2023: update copy for sharing feature

Fixes Automattic/martech#1501

## Proposed Changes

This PR is based on top of https://github.com/Automattic/wp-calypso/pull/74561.

* The current comparison grid contains 2 separate features: `Limited shares in social media` and `Unlimited shares in social media`. This change replaces these 2 features with a new feature, "Shares on social media" with plan-specific copy.

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/5436027/227462710-ace801d2-74bc-4e1c-8b5e-c9ef339ce569.png">
<img width="484" alt="image" src="https://user-images.githubusercontent.com/5436027/227462901-4f308141-7014-407e-bdf8-82bd3940e662.png">


_**Note: String Freeze label will be added after the review is complete.**_


## Testing Instructions



* Go to `/start/plans` and scroll down to the comparison grid.
* Confirm that under the "Growth and monetization tools" heading, the new feature is visible and matches the screenshot for both desktop and mobile.

## Pre-merge Checklist
